vaccine_feed_ingest/runners/pa/arcgis/normalize.py

```diff
@@ -28,7 +29,48 @@ def _get_id(site: dict) -> str:
     arcgis = "5b874ac0947347e9be49f6847eb44604"
     layer = 0
 
-    return f"{runner}:{site_name}:{arcgis}_{layer}:{data_id}"
+    return f"{runner}_{site_name}:{arcgis}_{layer}_{data_id}"
+
+
+def _get_website(site: dict) -> Optional[schema.Contact]:
+    if not site["attributes"]["Website"]:
+        return None
+    website_input = site["attributes"]["Website"].strip()
+    # check if the website field contains an email instead, for example
+    # "Social media/home page. Please contact Mktg Director first.last@example.com"
+    email_match = re.search(r"(?P<email>\S+@\S+)", website_input)
+    if email_match:
+        logger.warning(f"found email in website: '{website_input}'")
+        return schema.Contact(email=email_match.group("email"))
+
+    # if the input has any spaces after stripping it, this isn't a url we can parse
+    if re.search(r"\s", website_input):
+        logger.warning(f"Unable to parse website: '{website_input}'")
+        return None
+
+    website = re.sub(r"#.*", "", website_input)
+    # data observed in the wild may have a malformed scheme
+    website = re.sub(r"^(https?)//", r"\1://", website)
+    # check if the url scheme is missing
+    if re.match(r"^http", website):
+        return schema.Contact(website=website)
+
+    # try https://, otherwise add http://
+    website = "https://" + website_input
+    try:
+        requests.get(website, timeout=30)
+    except (
+        requests.exceptions.Timeout,
+        requests.exceptions.SSLError,
+        requests.ConnectionError,
+    ):
```

Comment:
I appreciate the desire to pass along accurate-but-SSL'd-if-possible protocols! That said, I think we should avoid making external requests in the `normalize` phase, and instead keep it as a pure data transformation.

With that said, could you use `http` for any of these that we're unconfident about? Most of them will probably upgrade sessions once they're started anyways.

CC: @bryanculbertson (in case you disagree)

Reply:
I guess it depends on the purpose of the website. If this is a site where users are going to have to enter sensitive identification info, I'd prefer to not risk a site not upgrading them to HTTPS, and risk comprimising their data. 
In an ideal world, all websites would have https and upgrade http to https, and so it would be fine to just add http and trust the website to Do The Rigth Thing... but I don't think we're there.

Reply:
We shouldn't call external services in the `parse` or `normalize` stage:
1. Both `parse` and `normalize` should be deterministic based on the input data alone. This helps when knowing when to re-run it.
2. We don't want to DDOS external services accidentally especially since there is no caching, and it is being run in automatically, in tests and on everyone's machines.
3. I want to eventually run `parse` and `normalize` in parallel in a env that may not have network access.

Validation and enrichment, like pre-upgrading http->https, could be an `enrich` step once I get a caching strategy working there. We run `enrich` for all locations in all feeds, so we could verify website protocols and accessibility in that stage.

For `normalize`, pick a reasonable default for the protocol. Perhaps https if all sites currently in the PA feed are https compatible?

Reply:
Ok, that makes sense. I'll stick with `http` so that we don't end up on a page that doesn't support `https` or has a outdated cert. It's better that users are able to get to the website than not, and an expired cert would throw up a scary look browser warning page that would deter folks.